adagrams/game.py

```diff
@@ -1,11 +1,168 @@
+            score += 8
+    for letter in word:
+        score += score_chart[letter.upper()]
+            
+    return score
+
+
+
+##-----------------------------------------##
+##-----------------WAVE 4------------------##
+##-----------------------------------------##
+"""
+4. You need a way to find the highest scoring word. This function looks at the list of word_list and calculates which 
+    of these words has the highest score, applies any tie-breaking logic, and returns the winning word in a special 
+    data structure.
+-Has one parameter: word_list, which is a list of strings
+-Returns a tuple that represents the data of a winning word and it's score. The tuple must contain the following elements:
+    -index 0 ([0]): a string of a word
+    -index 1 ([1]): the score of that word
+-In the case of tie in scores, use these tie-breaking rules:
+    -prefer the word with the fewest letters...
+    -...unless one word has 10 letters. If the top score is tied between multiple words and one is 10 letters long, 
+        choose the one with 10 letters over the one with fewer tiles
+    -If the there are multiple words that are the same score and the same length, pick the first one in the supplied list
+    reference: https://www.geeksforgeeks.org/python-difference-between-sorted-and-sort/
+    https://www.geeksforgeeks.org/python-sort-list-according-second-element-sublist/
+"""
+def get_highest_word_score(word_list):    
+
+
+    # word, pos = max(word_list, key=highest_score)
+    # tuple_highest_word.append(word, pos)
+    # return tuple_highest_word
+# return max(word_list, key=get_value)
+# def get_value(word):
+#     """Return word letter score."""
+#     return sum(score_chart[letter] for letter in word)
+
+    score = []
+    highest_scores = []
+
+    for i in range(len(word_list)):
+        if i > 0:
+            if score_word(word_list[i]) > score[1]:
+                highest_scores.clear()
+                score = [word_list[i], score_word(word_list[i])]
+                highest_scores.append(score)
+            elif score_word(word_list[i]) == score[1]:
+                score = [word_list[i], score_word(word_list[i])]
+                highest_scores.append(score)
+            else:
+                continue
+        else:
+            score = [word_list[i], score_word(word_list[i])]
+            highest_scores.append(score)
 
-def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    if len(highest_scores) == 1: # highest_score = [[word_list[i], score_word(word_list[i])]]
+        return highest_scores[0]
+    elif len(highest_scores) > 1:
+        print(f"highest scores before sort {highest_scores}")
+        highest_scores = sorted(highest_scores, key = lambda x: len(x[0]))
+        print(f"highest scores after sort {highest_scores}")
+        for i in range(len(highest_scores)):
+            if len(highest_scores[i][0]) >= 10:
+                return highest_scores[i]
+        return highest_scores[0]
```

Comment:
All in all, this code works, but it is easy to get lost in terms of readability since you are constantly updating and checking lists. Reworking the algorithm here is the easiest way to work with it. I think where y'all got a little tripped up was in using the lists over a tuple. A more concise algorithm would be as follows:

     1) Create a tuple to hold the highest word and give it default values of an empty string and 0.
     2) For each word in the word list:
          a) Get its score and store it
          b) if the score is greater than index 1 of your highest word tuple, replace the highest word tuple with the new 
          word and score.
          c) If the score is equal to index 1 of the tuple, run the following checks:
               i) If the length of the current highest word is equal to 10, continue
               ii) Else if the length of the word is 10 and the current highest word is not, replace the highest word with the 
               word you are currently assessing.
               iii) Else if the length of your current word is less than the length of the highest word, replace the highest 
               word with the word you are currently assessing.